pulseapi/profiles/views.py

```diff
@@ -66,3 +67,50 @@ def put(self, request, *args, **kwargs):
             pass
 
         return super(UserProfileAPIView, self).put(request, *args, **kwargs)
+
+
+class ProfileCustomFilter(filters.FilterSet):
+    """
+      We add custom filtering to allow you to filter by:
+
+      * Profile type - pass the `?profile_type=` query parameter
+      * Program type - pass the `?program_type=` query parameter
+      * Program year - pass the `?program_year=` query parameter
+    """
+    profile_type = django_filters.CharFilter(
+        name='profile_type__value',
+        lookup_expr='iexact',
+    )
+    program_type = django_filters.CharFilter(
+        name='program_type__value',
+        lookup_expr='iexact',
+    )
+    program_year = django_filters.CharFilter(
+        name='program_year__value',
+        lookup_expr='iexact',
+    )
+
+    class Meta:
+        """
+        Required Meta class
+        """
+        model = UserProfile
+        fields = [
+            'profile_type',
+            'program_type',
+            'program_year',
+        ]
+
+
+class UserProfileAPISearchView(ListCreateAPIView):
```

Comment:
I don't think this should allow `Create` on this view since we don't have any logic to handle that. Can we just have it be a simple list view?
```py
class UserProfilePublicListAPIView(ListAPIView):
    ...
```

Reply:
yarr, good catch